src/utils/num.ts

```diff
@@ -181,3 +181,27 @@ export function addPercent(number: BigNumberish, percent: number) {
   const bigIntNum = BigInt(number);
   return bigIntNum + (bigIntNum * BigInt(percent)) / 100n;
 }
+
+/**
+ * Check if a value is a number.
+ *
+ * @param {unknown} value - The value to check.
+ * @return {boolean} Returns true if the value is a number, otherwise returns false.
+ */
+export function isNumber(value: unknown): value is number {
+  return typeof value === 'number';
+}
+
+/**
+ * Checks if a given value is a string.
+ *
+ * @param {unknown} value - The value to be checked.
+ * @return {boolean} - Returns true if the value is a string, false otherwise.
+ */
+export function isString(value: unknown): value is string {
+  return typeof value === 'string';
+}
+
+export function isBoolean(value: unknown): value is boolean {
```

Comment:
nitpicks: 

- JSDoc
-  We do not have a Miscellaneous utils file so there is no perfect spot for these two 'helper' methods, bool is kinda ok here but to import isString from num seems kinda strange, I would suggest using shortString es not perfect alternative but seems more natural.

Reply:
@ikrcatov so please add on these functions JSDoc-s - which will automatically generate documentation !
And for second point, maybe move `isString` to the shortString util file 🙏

Reply:
Sure